src/client/shader.cpp

```diff
@@ -140,8 +141,25 @@ class SourceShaderCache
 			type = "geometry";
 		else if (str_ends_with(filename, ".fs.glsl"))
 			type = "fragment";
-		else
+		else {
 			errorstream << "Unsupported shader type: " << filename << std::endl;
+			return;
+		}
+		if (program.length() > 10000) {
+			errorstream << "Rejecting long shader (> 10k chars): " << filename << std::endl;
```

Comment:
The standard node shader is 6.5kB, so the limit is definitely too low. And why do you need a special limit for shaders?

Reply:
Increased limit to 65kB. This is just to prevent extremely large shaders being sent, intended as some sort of cheap DOS-protection...